multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/ServiceStep.java

```diff
@@ -67,15 +67,15 @@ protected abstract OperationExecutionState executeOperation(ProcessContext conte
     protected void processServiceActionFailure(ProcessContext context, CloudServiceInstanceExtended serviceInstance,
                                                CloudOperationException e) {
         if (!serviceInstance.isOptional()) {
-            String detailedDescription = MessageFormat.format(Messages.ERROR_CREATING_SERVICE, serviceInstance.getName(),
-                    serviceInstance.getLabel(), serviceInstance.getPlan(), e.getDescription());
             if (e.getStatusCode() == HttpStatus.BAD_GATEWAY) {
                 context.setVariable(Variables.SERVICE_OFFERING, serviceInstance.getLabel());
-                throw new CloudServiceBrokerException(e.getStatusCode(), e.getStatusText(), detailedDescription);
+                throw new CloudServiceBrokerException(e.getStatusCode(), e.getStatusText(), e.getDescription());
             }
-            throw new CloudControllerException(e.getStatusCode(), e.getStatusText(), detailedDescription);
+            throw new CloudControllerException(e.getStatusCode(), e.getStatusText(), e.getDescription());
         }
-        getStepLogger().warn(e.getDescription());
+        String detailedMessage = serviceInstance.isOptional() ? Messages.ERROR_CREATING_OR_UPDATING_OPTIONAL_SERVICE_INSTANCE
+            : Messages.ERROR_CREATING_OR_UPDATING_SERVICE_INSTANCE;
```

Comment:
Always optional true. Add the other message for not optional above
Use placeholders "{0}" instead of concatenation